gtsfm/ui/process_graph_generator.py

```diff
@@ -57,6 +58,11 @@ def _build_graph(self) -> None:
         sorted_metadata = sorted(list(unique_metadata))
 
         for metadata in sorted_metadata:
+            if self.is_image_correspondence and metadata.display_name in ["Matcher", "DetectorDescriptor"]:
```

Comment:
ideally I dont think we want to hard-code these in the graph generation code. 
These will break if we change the code.

Reply:
@dellaert WDYT? 
Everything else LGTM in the PR.

Reply:
@akshay-krishnan I agree that this might cause difficulties if we change the code. Do you think it would be better if I change the second part of the condition with `metadata.parent_plate == "DetDescCorrespondenceGenerator"`?

This way if we change/add new nodes as long as the parent_plate is the same the code will choose the right pathway for the graph.

Reply:
Yeah, we can expect the parent plate to change less often than the blue nodes.

Reply:
it may not change at all, but also could at some point..